### PR TITLE
feat(search): add search

### PR DESCRIPTION
**Pull request by oysteinsigholt**

### CODE REVIEW

File: docs/features/syntaxHighlight.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+local var = "untyped code"
+```
+
+```js
+var t = 'javascript code'
+```
+
+``` http
```
Comment (oysteinsigholt) [inline docs/features/syntaxHighlight.md]: The goal was to demonstrate that both syntaxes work. I should add some explaining text at least.

File: package.json [hunk 2]
```diff
@@ -5,13 +5,13 @@
   "main": "index.js",
   "scripts": {
     "clean": "del-cli build/",
```
Comment (oysteinsigholt) [inline package.json]: Will fix (see #18). I landed on del-cli for no particular reason, but I see no reason to change to rimraf unless we gain something from it.

File: script/script.js [hunk 3]
```diff
@@ -1,18 +1,178 @@
...
         }
     });
+
     if (typeof pageReady === "function") {
         pageReady(document);
     }
+
+    loadSearchIndex();
+
+    // search close button
```
Comment (oysteinsigholt) [inline script/script.js]: 1. Will do.
2. `DOMContentLoaded` fires only on a clean page reload, while `newPageReady` fires on every Barba.js page transition. There should be no event listeners being set up in `newPageReady`, am I missing something?

Comment (jorunfa) [inline script/script.js]: 2. No I'm misreading the indentations here. Sorry. I thought the function calls below were inside the `newPageReady` event function.

File: lib/build-docs.js [hunk 4]
```diff
@@ -53,6 +54,19 @@ metalsmith(path.dirname(__dirname))
...
+    .use(lunr({
+        ref: "url",
+        indexPath: "index.json",
+        fields: {
+            plaintext: 1,
+            title: 5,
+            tags: 10
+        },
+        includeFields: true,
+        preprocess: (content) => {
```
Comment (oysteinsigholt) [inline lib/build-docs.js]: A preprocess function that does nothing us unnecessary, and should be removed.

**Comment by jorunfa** [comment]
Good enough, I think.

**Pull request merged**